### PR TITLE
Bump version to 3.1.0 and improve changelog

### PR DESCRIPTION
**Pull request by enrico-usai**
Reordered changelog, added items from cookbook and entry about clusters in subnets with no internet access.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,28 +5,37 @@ CHANGELOG
...
-- Add support for `UseEc2Hostnames` in the cluster configuration file. When set to `true`, use EC2 default hostnames (e.g. ip-1-2-3-4) for compute nodes.
-- Explicitly set cloud-init datasource to be EC2. This save boot time for Ubuntu and CentOS platforms.
-- Add support for multiple compute resources with same instance type per queue.
-- Add `parallelcluster:compute-resource-name` tag to LaunchTemplates used by compute nodes.
 - Add abbreviated flags for `cluster-name` (-n), `region` (-r), `image-id` (-i) and `cluster-configuration` / `image-configuration` (-c) to the cli.
 - Enable clusters to authenticate users by integrating with Active Directory (AD) domains managed via AWS Directory Service.
+- Enable cluster creation in subnets with no internet access.
+- Add support for multiple compute resources with same instance type per queue.
+- Add support for `UseEc2Hostnames` in the cluster configuration file. When set to `true`, use EC2 default hostnames (e.g. ip-1-2-3-4) for compute nodes.
+- Install NVIDIA and CUDA for ARM.
```
Comment (lukeseawalker) [inline CHANGELOG.md]: probably better to specify "NVIDIA driver and CUDA library"

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,28 +5,37 @@ CHANGELOG
...
+- Use compute resource name rather than instance type in compute fleet Launch Template name.
+- Enable possibility to suppress `SlurmQueues` and `ComputeResources` length validators.
 - Disable EC2 ImageBuilder enhanced image metadata when building ParallelCluster custom images.
 - Disable packages update at instance launch time on Amazon Linux 2.
 
 **BUG FIXES**
-- Redirect stderr and stdout to CLI log file to prevent unwanted text to pollute the pcluster CLI output.
-- Fix ecs:ListContainerInstances permission in BatchUserRole
+- Redirect stderr and stdout to CLI log file to prevent unwanted text to pollute the `pcluster` CLI output.
+- Fix `ecs:ListContainerInstances` permission in BatchUserRole.
```
Comment (gmarciani) [inline CHANGELOG.md]: Shouldn't also BatchUserRole be enclosed in back ticks?

**Review by gmarciani — DISMISSED** [review]

**Review by gmarciani — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,28 +5,37 @@ CHANGELOG
 ------
 
 **ENHANCEMENTS**
-- Add support for `UseEc2Hostnames` in the cluster configuration file. When set to `true`, use EC2 default hostnames (e.g. ip-1-2-3-4) for compute nodes.
-- Explicitly set cloud-init datasource to be EC2. This save boot time for Ubuntu and CentOS platforms.
-- Add support for multiple compute resources with same instance type per queue.
-- Add `parallelcluster:compute-resource-name` tag to LaunchTemplates used by compute nodes.
 - Add abbreviated flags for `cluster-name` (-n), `region` (-r), `image-id` (-i) and `cluster-configuration` / `image-configuration` (-c) to the cli.
 - Enable clusters to authenticate users by integrating with Active Directory (AD) domains managed via AWS Directory Service.
```
Comment (mauri-melato) [inline CHANGELOG.md:9]: "Add support for multiple users cluster environments by integrating with..."
Also this should be at the top of the list

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,28 +5,37 @@ CHANGELOG
...
+- Enable cluster creation in subnets with no internet access.
```
Comment (mauri-melato) [inline CHANGELOG.md:10]: This comes second

**Pull request merged**